skimage/filters/thresholding.py

```diff
@@ -271,16 +271,14 @@ def threshold_otsu(image, nbins=256):
     -----
     The input image must be grayscale.
     """
-    if len(image.shape) > 2 and image.shape[-1] in (3, 4):
+    if image.ndim > 2 and image.shape[-1] in (3, 4):
         msg = "threshold_otsu is expected to work correctly only for " \
               "grayscale images; image shape {0} looks like an RGB image"
         warn(msg.format(image.shape))
 
     # Check if the image is multi-colored or not
-    if image.min() == image.max():
-        raise ValueError("threshold_otsu is expected to work with images "
-                         "having more than one color. The input image seems "
-                         "to have just one color {0}.".format(image.min()))
+    if np.all(image == image[0]):
```

Comment:
I prefer: 

```suggestion
    if np.all(image == image.ravel()[0]):
```

But it's a matter of taste. =)

Reply:
You are right @jni, I used to prefer `ravel`, but according to the _Notes_ section in its [doc](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html)
> When a view is desired in as many cases as possible, arr.reshape(-1) may be preferable.

Otherwise `ravel` is super fine ^^

Reply:
@rfezzani 

> When a view is desired in as many cases as possible, arr.reshape(-1) may be preferable.

You already taught me this once before in the `array_map` PR and I've already forgotten it! 😅 Old habits die hard... Thanks!